salesforce/my-conservation-life/force-app/main/default/lwc/controllers/__tests__/projectsController.test.js

```diff
@@ -0,0 +1,117 @@
+import utils from 'c/utils';
+import projects from '../projectsController';
+
+const querystring = require('querystring');
+
+const createMockFetch = (jsonObject) => jest.fn(() => Promise.resolve({ ok: true, json: () => jsonObject }));
+
+let PROJECTS_URL = new URL(utils.URL);
+PROJECTS_URL.pathname += '/projects';
+
+describe('projects.find', () => {
+    
+    const EXPECTED_PROJECTS = [];
+    let fetch;
+
+    beforeEach(() => {
+        fetch = createMockFetch(EXPECTED_PROJECTS);
+        global.fetch = fetch;
+    });
+
+    it('finds all', async () => {
+        const assetArray = await projects.find();
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);
+    });
+
+    it('finds without Project ID', async () => {
+        // name and sponsor id
+        const params = {sponsor_id: '2', name: 'foo'};
+        const query = querystring.encode(params);
+        const assetArray = await projects.find(params);
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href + `?${query}`);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);    
+    });
+
+    it('finds without Sponsor ID', async () => {
+        // name and project id
+        const params = {id: '2', name: 'foo'};
+        const query = querystring.encode(params);
+        const assetArray = await projects.find(params);
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href + `?${query}`);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);
+    });
+
+    it('finds without Project Name', async () => {
+        // sponsor id and project id
+        const params = {id: '1', sponsor_id: '2'};
+        const query = querystring.encode(params);
+        const assetArray = await projects.find(params);
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href + `?${query}`);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);   
+    });
+
+    it('finds with Project Name only', async () => {
+        // name
+        const params = {name: 'foo'};
+        const query = querystring.encode(params);
+        const assetArray = await projects.find(params);
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href + `?${query}`);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);  
+    });
+
+    it('finds with Sponsor ID only', async () => {
+        // sponsor id
+        const params = {sponsor_id: '1'};
+        const query = querystring.encode(params);
+        const assetArray = await projects.find(params);
+        expect(fetch.mock.calls[0][0]).toBe(PROJECTS_URL.href + `?${query}`);
+        expect(assetArray).toEqual(EXPECTED_PROJECTS);  
+    });
+
+    it('finds with Project ID only', async () => {
```

Comment:
All of these 'it' blocks (aside from the first) appear to be doing pretty similar things - looks like only 1 line is different between each of them. Would it make sense to refactor it into one method and make the line change a passable variable from the it block? I can explain better in person if that doesn't make sense.